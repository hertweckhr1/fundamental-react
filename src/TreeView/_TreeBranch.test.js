import React from 'react';
import renderer from 'react-test-renderer';
import TreeBranch from './_TreeBranch';
import TreeView from './TreeView';

// mock shortid for snapshot testing
jest.mock('shortid', () => {
    let id = 1;

    return {
        generate: () => id++
    };
});

describe('<TreeBranch />', () => {
    const treeBranch = (
        <TreeBranch>
            <TreeView.Item>
                <TreeView.Row>
                    <TreeView.Col>GreatGrandchild 1</TreeView.Col>
                </TreeView.Row>
            </TreeView.Item>
        </TreeBranch>
    );

    test('create treebranch component', () => {
        let component = renderer.create(treeBranch);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
