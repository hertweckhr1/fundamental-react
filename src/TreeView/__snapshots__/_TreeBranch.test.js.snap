// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeBranch /> create treebranch component 1`] = `
<ul
  aria-hidden={false}
  className=""
  role="group"
>
  <li
    aria-expanded={false}
    className="fd-tree__item"
    id="fd-1"
    role="treeitem"
  >
    <div
      className="fd-tree__row"
    >
      <div
        className="fd-tree__col fd-tree__col--control"
      >
        GreatGrandchild 1
      </div>
    </div>
  </li>
</ul>
`;
